### DISCUSSION TITLE
Is it possible to delete the installed old version of the plugin after updating the plugin version to save space

### DISCUSSION THREAD

**Discussion by czfadmin** [microsoft/vscode-discussions#563] [Extension Development QnA] [answered]
The installed plug-in has been updated. The user chooses to update the plug-in, but why not delete the old version of the plug-in in the plug-in folder. Deleting the old version can save disk space. What is the intention of not deleting the old version of the plug-in?

**Comment by r3m0t** [comment, marked as answer]
Hi, as I understand it  the old extension folder name is written to a file `.obsolete` and is deleted when VS Code is next opened (from a cold start, ie when no windows are currently open). This is because old windows can still be using the extension from the old folder. If you are seeing something else I suggest you open a bug report.

**Reply by czfadmin** [reply]
> Hi, as I understand it the old extension folder name is written to a file `.obsolete` and is deleted when VS Code is next opened (from a cold start, ie when no windows are currently open). This is because old windows can still be using the extension from the old folder. If you are seeing something else I suggest you open a bug report.

Thank you for your answer, I will follow the steps you said later to see if there is such a situation
